### PR TITLE
[quant][ao_migration] Base package in tests

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77638
* #77630
* #77627
* #77586
* #77582
* #77581
* __->__ #77064
* #77344
* #77065

Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

Differential Revision: [D36459452](https://our.internmc.facebook.com/intern/diff/D36459452)

### CODE REVIEW

File: test/quantization/ao_migration/common.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 
 class AOMigrationTestCase(TestCase):
-    def _test_package_import(self, package_name: str):
+    def _test_package_import(self, package_name: str, base: Optional[str] = None):
         r"""Tests the module import by making sure that all the internals match
         (except the dunder methods)."""
-        old_module = importlib.import_module(f'torch.quantization.{package_name}')
-        new_module = importlib.import_module(f'torch.ao.quantization.{package_name}')
+        if base is None:
+            base = 'quantization'
```
Comment (jerryzh168) [inline test/quantization/ao_migration/common.py:11]: str is not mutable in python so it's safe to use as a default arg?

Comment (jerryzh168) [reply test/quantization/ao_migration/common.py:11]: same for other two functions

**Review by jerryzh168 — APPROVED** [review]
had an inline nit comment

**Comment by z-a-f** [comment]
@z-a-f has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36459452).

**Pull request closed without merging**